File: Challenge 27/c27_easy.py
'''
    Daily Programmer Challenge 27 : Easy
    @author:    Baron Daugherty
    @date:      2015-07-20
    @desc:      Write a program that accepts a year as input and outputs the
                century the year belongs in (e.g. 18th century's year ranges are
                1701-1800) and whether or not the year is a leapyear.
'''
from datetime import date

year = int(input("Please enter a year A.D.: "))
d = date(year, 1, 1)

print("Year: " +str(year))
if year % 100 == 0:
    print(str(int(year/100)) + " Century")
else:
    print(str(int(year/100 +1)) +" Century")

if year % 4 != 0:
    print(str(year) +" is not a leap year.")
elif year % 100 != 0:
    print(str(year) +" is not a leap year.")
elif year % 400 != 0:
    print(str(year) +" is not a leap year.")
else:
    print(str(year) +" is a leap year")

File: Challenge 27/c27_easy.cpp
/*
	Daily Programmer Challenge 27 : Easy
	@author:	Baron Daugherty
	@date:		2015-09-18
	@desc:		Input a year and output the century it is in.
				Also output if it is/is not a leap year.
*/

/* INCLUDES and NAMEPSACES */
#include<iostream>
using namespace std;

/* FUNCTIONS */
int main()
{
	//needed variables
	uint16_t year;
	uint16_t century;
	bool leap;

	//get the year
	cout << "Please enter a year A.D.: ";
	cin >> year;

	//calculate the century
	if (year % 100 == 0)
		century = year / 100;
	else
		century = (year / 100) + 1;

	//determine if leap year
	if (((year % 4 == 0) && (year % 100 != 0))
		|| (year % 400 == 0))
		leap = true;
	else
		leap = false;

	//output
	cout << year << " is in the " << century << " century.\n";
	cout << "It" << (leap ? " is " : " is not ") << "a leap year.\n";

	//hunky-dory
	system("pause");
	return 0;
}	//END main